README.md
## uxcore-layout


[![NPM version][npm-image]][npm-url]
[![build status][travis-image]][travis-url]
[![Test Coverage][coveralls-image]][coveralls-url]
[![Dependency Status][dep-image]][dep-url]
[![devDependency Status][devdep-image]][devdep-url]
[![NPM downloads][downloads-image]][npm-url]

[![Sauce Test Status][sauce-image]][sauce-url]

[npm-image]: http://img.shields.io/npm/v/uxcore-layout.svg?style=flat-square
[npm-url]: http://npmjs.org/package/uxcore-layout
[travis-image]: https://img.shields.io/travis/uxcore/uxcore-layout.svg?style=flat-square
[travis-url]: https://travis-ci.org/uxcore/uxcore-layout
[coveralls-image]: https://img.shields.io/coveralls/uxcore/uxcore-layout.svg?style=flat-square
[coveralls-url]: https://coveralls.io/r/uxcore/uxcore-layout?branch=master
[dep-image]: http://img.shields.io/david/uxcore/uxcore-layout.svg?style=flat-square
[dep-url]: https://david-dm.org/uxcore/uxcore-layout
[devdep-image]: http://img.shields.io/david/dev/uxcore/uxcore-layout.svg?style=flat-square
[devdep-url]: https://david-dm.org/uxcore/uxcore-layout#info=devDependencies
[downloads-image]: https://img.shields.io/npm/dm/uxcore-layout.svg
[sauce-image]: https://saucelabs.com/browser-matrix/uxcore-layout.svg
[sauce-url]: https://saucelabs.com/u/uxcore


#### setup develop environment

```sh
$ git clone https://github.com/uxcore/uxcore-layout
$ cd uxcore-layout
$ npm install
$ npm start
```

## Usage

```javascript
let Layout = require('../src');
let {Left, Right} = Layout;
let classnames = require('classnames');

class Demo extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return <div>
            <p>左侧自适应，右侧固定</p>
            <Layout className="layoutDemo">
                <Left adaptive={true} className="left">左</Left>
                <Right width={190} className="right">右</Right>
            </Layout>
            <p>右侧自适应，左侧固定</p>
            <Layout className="layoutDemo">
                <Left width={190} className="left">左</Left>
                <Right adaptive={true} className="right">右</Right>
            </Layout>
            <p>左右都固定</p>
            <Layout className="layoutDemo">
                <Left width={500} className="left">左</Left>
                <Right width={500} className="right">右</Right>
            </Layout>
        </div>
        
    }
}

module.exports = Demo;
```

## Props

| 配置项 | 类型 | 必填 | 默认值 | 功能/备注 |
|---|---|---|---|---|
|className|string|optional|""|加入额外的类名，在使用 kuma 的基础上进行适当的定制时会用得到|


## 子组件

### Left/Right

| 配置项 | 类型 | 必填 | 默认值 | 功能/备注 |
|---|---|---|---|---|
|className|string|optional| |加入额外的类名，在使用 kuma 的基础上进行适当的定制时会用得到|
|width|number|optional|500|布局块的宽度，仅在定宽时起作用|
|adaptive|boolean|optional| |是否为自适应，未设定 adaptive=true 的一侧，则为定宽，不可两者皆为自适应|




HISTORY.md
# History

## 1.2.0
`CHANGED` support js style export

## 1.0.8

`CHANGE` export ES Module 

## 1.0.7

`CHANGE` adapted React 15.x

## 1.0.5

`CHANGED` update scaffold

## 1.0.4

`CHANGED` change repository to https

## 1.0.3

`FIXED` fix issue #2

## 1.0.2

`FIXED` fix issue #1 



demo/LayoutDemo.js
let Layout = require('../src');
let { Left, Right } = Layout;
let Menu = require('uxcore-menu');
let Form = require('uxcore-form');
let Button = require('uxcore-button');
const React = require('react');

let SubMenu = Menu.SubMenu;
let {
  InputFormField,
  FormRow,
  OtherFormField,
  DateFormField,
  SelectFormField,
  CheckboxGroupFormField,
} = Form;
let Item = CheckboxGroupFormField.Item;
const classnames = require('classnames');

class Demo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      current: '1',
    };
  }

  componentDidMount() {
    const me = this;
    $('.J_enter').on('keydown', function (e) {
      if (e.which === 13) {
        console.log(me.refs.form.getValues());
      }
    });
  }

  componentWillUnmount() {
    $('.J_enter').off('keydown');
  }

  handleFRClick(from) {
    alert(from);
  }

  handleClick(e) {
    console.log('click ', e);
    this.setState({
      current: e.key,
    });
  }

  handleSearchClick() {
    const me = this;
    me.setState({
      showAdSearch: !me.state.showAdSearch,
    });
  }

  handleSubmitClick() {
    const me = this;
    console.log(me.refs.form.getValues());
  }

  handleResetClick() {
    const me = this;
    me.refs.form.resetValues();
  }

  render() {
    const me = this;
    return (<div>
      <p>默认</p>
      <Layout>
        <Left/>
        <Right/>
      </Layout>
      <p>左侧自适应，右侧固定</p>
      <Layout className="layoutDemo">
        <Left adaptive className="left">
          <span onClick={me.handleFRClick.bind(me, 'left')}>左</span>
        </Left>
        <Right width={190} className="right">
          <span onClick={me.handleFRClick.bind(me, 'right')}>右</span>
        </Right>
      </Layout>
      <p>右侧自适应，左侧固定</p>
      <Layout className="layoutDemo">
        <Left width={190} className="left">
          <span onClick={me.handleFRClick.bind(me, 'left')}>左</span>
        </Left>
        <Right adaptive className="right">
          <span onClick={me.handleFRClick.bind(me, 'right')}>右</span>
        </Right>
      </Layout>
      <p>左右都固定</p>
      <Layout className="layoutDemo fixed">
        <Left width={500} className="left">
          <span onClick={me.handleFRClick.bind(me, 'left')}>左</span>
        </Left>
        <Right width={500} className="right">
          <span onClick={me.handleFRClick.bind(me, 'right')}>右</span>
        </Right>
      </Layout>
      <p>实际场景</p>
      <Layout className="demo4">
        <Left width={190}>
          <Menu onClick={this.handleClick.bind(this)}
            style={{ width: 190 }}
            defaultOpenKeys={['sub1']}
            selectedKeys={[this.state.current]}
            mode="inline"
          >
            <SubMenu
              key="sub1"
              title={<span><i className="kuma-icon kuma-icon-email"></i><span>导航一</span></span>}
            >
              <Menu.Item key="1">选项1</Menu.Item>
              <Menu.Item key="2">选项2</Menu.Item>
              <Menu.Item key="3">选项3</Menu.Item>
              <Menu.Item key="4">选项4</Menu.Item>
            </SubMenu>
            <SubMenu
              key="sub2"
              title={<span><i className="kuma-icon kuma-icon-wangwang"></i><span>导航二</span></span>}
            >
              <Menu.Item key="5">选项5</Menu.Item>
              <Menu.Item key="6">选项6</Menu.Item>
              <SubMenu key="sub3" title="三级导航">
                <Menu.Item key="7">选项7</Menu.Item>
                <Menu.Item key="8">选项8</Menu.Item>
              </SubMenu>
            </SubMenu>
            <SubMenu
              key="sub4"
              title={<span><i className="kuma-icon kuma-icon-boss"></i><span>导航三</span></span>}
            >
              <Menu.Item key="9">选项9</Menu.Item>
              <Menu.Item key="10">选项10</Menu.Item>
              <Menu.Item key="11">选项11</Menu.Item>
              <Menu.Item key="12">选项12</Menu.Item>
            </SubMenu>
          </Menu>
        </Left>
        <Right adaptive>
          <Form ref="form">
            <FormRow>
              <InputFormField
                className="J_enter"
                jsxplaceholder="输入 xxxx 进行查询"
                jsxshowLabel={false} jsxname="search"
              />
              <OtherFormField jsxflex={2}>
                <div className="demo-updown" onClick={me.handleSearchClick.bind(me)}>
                  <a href="javascript:void(0);">高级查询</a>
                  <i
                    className={classnames({
                      'kuma-icon': true,
                      'kuma-icon-title-up': me.state.showAdSearch,
                      'kuma-icon-title-down': !me.state.showAdSearch,
                    })}
                  ></i>
                </div>
              </OtherFormField>
            </FormRow>
            <FormRow
              className={classnames({
                'f-dn': !me.state.showAdSearch,
              })}
            >
              <InputFormField jsxlabel="名称" jsxname="input1" />
              <DateFormField
                jsxlabel="名称"
                jsxname="date1"
                jsxtype="cascade"
                style={{
                  width: 100,
                }}
              />
            </FormRow>
            <FormRow
              className={classnames({
                'f-dn': !me.state.showAdSearch,
              })}
            >
              <SelectFormField
                className="select2"
                jsxfetchUrl="http://suggest.taobao.com/sug"
                jsxmultiple
                jsxname="select2"
                jsxlabel="名称"
                afterFetch={(obj) => {
                  const data = {};
                  obj.result.forEach((item) => {
                    data[item[1]] = item[0];
                  });
                  console.log(data);
                  return data;
                }}
              />
              <CheckboxGroupFormField jsxlabel="名称" jsxname="check1">
                <Item value="1" text="名称" />
                <Item value="2" text="名称" />
                <Item value="3" text="名称" />
              </CheckboxGroupFormField>
            </FormRow>
            <FormRow
              className={classnames({
                'f-dn': !me.state.showAdSearch,
              })}
            >
              <OtherFormField className="button1">
                <Button type="primary" onClick={me.handleSubmitClick.bind(me)}>提交</Button>
                <Button type="secondary" onClick={me.handleResetClick.bind(me)}>重置</Button>
              </OtherFormField>
            </FormRow>
          </Form>
        </Right>
      </Layout>
    </div>);
  }
}

module.exports = Demo;
